tikv/region.go

```diff
@@ -313,6 +316,25 @@ func (rm *RaftRegionManager) OnRegionConfChange(ctx *raftstore.PeerEventContext,
 	region.updateRegionEpoch(epoch)
 }
 
+func (rm *RaftRegionManager) OnRoleChange(ctx *raftstore.PeerEventContext, newState raft.StateType) {
+	rm.mu.RLock()
+	region, ok := rm.regions[ctx.RegionId]
+	rm.mu.RUnlock()
+	if ok {
+		if bytes.Compare(region.startKey, []byte{}) == 0 {
+			newRole := Follower
+			if newState == raft.StateLeader {
+				newRole = Leader
+			}
+			log.Infof("[ray] first region role change newRole=%v", newRole)
+			rm.mvccStore.deadlockDetector.RoleChange(int32(newRole))
+		}
+	} else {
+		// should not be here
+		log.Warnf("[ray] regionid=%v not found", ctx.RegionId)
```

Comment:
Is it a bug if the region map does not contain the event `regionId` ? @coocood

Reply:
Yes.